rover/__tests__/handler/transaction.test.ts

```diff
@@ -0,0 +1,129 @@
+// import {
+//   filterTransaction,
+//   getTransactionData,
+//   transactionHandler,
+// } from '../../src/handler/transaction'
+
+import Transaction from '../../src/handler/transaction'
+import { Registry } from '../../src/registry'
+
+describe('getTransactionData', () => {
+  it('requests transaction details from the provider', () => {
+    expect.assertions(1)
+    let provider = {
+      getTransaction: jest.fn(),
+    }
+
+    let transactionHash
+
+    Transaction.getTransactionData(provider, transactionHash)
+    expect(provider.getTransaction).toHaveBeenCalled()
+  })
+})
+
+describe('filterTransaction', () => {
+  describe('when the transction to address is included in the registry', () => {
+    it('returns true', async () => {
+      expect.assertions(1)
+      Registry.get = jest.fn().mockResolvedValue(['0xabc'])
+
+      let transaction = { to: '0xabc' }
+      let connection
+
+      expect(await Transaction.filterTransaction(transaction, connection)).toBe(
+        true
+      )
+    })
+  })
+
+  describe('when the transaction from address is included in the registry', () => {
+    it('returns true', async () => {
+      expect.assertions(1)
+      Registry.get = jest.fn().mockResolvedValue(['0xabc'])
+
+      let transaction = { from: '0xabc' }
+      let connection
+
+      expect(await Transaction.filterTransaction(transaction, connection)).toBe(
+        true
+      )
+    })
+  })
+
+  describe('when the sender and recipient are not included in the trasnaction', () => {
+    it('returns false', async () => {
+      expect.assertions(1)
+      Registry.get = jest.fn().mockResolvedValue(['0xabc'])
+      let transaction = { to: '0xbbb' }
+      let connection
+
+      expect(await Transaction.filterTransaction(transaction, connection)).toBe(
+        false
+      )
+    })
+  })
+})
+
+describe('transactionHandler', () => {
+  describe('when the transaction hash exists', () => {
+    describe('when the transaction is relevant', () => {
```

Comment:
can you be more specific about what makes a transaction relevant in the test description?